File: frontend/src/app/dto/Filter.ts
export class Filter {
  public highestPrice: number;
  public lowestPrice: number;
  public orderByPrice: boolean;

  constructor(highestPrice: number, lowestPrice: number, orderByPrice: boolean) {
    this.highestPrice = highestPrice;
    this.lowestPrice = lowestPrice;
    this.orderByPrice = orderByPrice;
  }
}

File: src/main/java/coursework/prostranstvo/model/dto/Filter.java
package coursework.prostranstvo.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Optional;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Filter {
    private Double highestPrice;
    private Double lowestPrice;
    //true - ascending, false - descending
    private Optional<Boolean> orderByPrice;
}
